core/src/test/java/brooklyn/location/basic/ByonLocationResolverTest.java

```diff
@@ -307,12 +308,41 @@ public void testEmptySpec() throws Exception {
                 "name", "foo",
                 "user", "myuser"
         );
-        MachineProvisioningLocation<SshMachineLocation> provisioner = resolve(spec, flags);
-        SshMachineLocation location1 = provisioner.obtain(ImmutableMap.of());
+        MachineProvisioningLocation<MachineLocation> provisioner = resolve(spec, flags);
+        SshMachineLocation location1 = (SshMachineLocation)provisioner.obtain(ImmutableMap.of());
         Assert.assertEquals("myuser", location1.getUser());
         Assert.assertEquals("1.1.1.1", location1.getAddress().getHostAddress());
     }
 
+    @Test
+    public void testWindowsMachines() throws Exception {
+        brooklynProperties.put("brooklyn.location.byon.windows.username", "myuser");
+        brooklynProperties.put("brooklyn.location.byon.windows.password", "mypassword");
+        String spec = "byon";
+        Map<String, ?> flags = ImmutableMap.of(
+                "hosts", ImmutableList.of("1.1.1.1", "2.2.2.2"),
+                "osfamily", "windows"
+        );
+        MachineProvisioningLocation<MachineLocation> provisioner = resolve(spec, flags);
+        MachineLocation location = provisioner.obtain(ImmutableMap.of());
+
+        assertEquals(location.config().get(WinRmMachineLocation.WINDOWS_USERNAME), "myuser");
+        assertEquals(location.config().get(WinRmMachineLocation.WINDOWS_PASSWORD), "mypassword");
+        Assert.assertNotNull(provisioner);
```

Comment:
why assertNotNull here? We've already called methods on it. Suggest delete this line?